src/code_translation/ir/nodes.py

```diff
@@ -100,6 +113,15 @@ def sql_string(self) -> Optional[str]:
         left_set_key = False
         if isinstance(self.left, SQLNode):
             left_sql = self.left.sql_string.strip('"').strip("'")
+        elif isinstance(self.left, JoinNode):
+            left_sql = self.left.sql_string.strip('"').strip("'")
+        elif isinstance(self.left, AggregationNode):
+            # This is where the dependency injection starts to suck...
+            # We just want to find out if there is a precode condition for the CSTTranslation
+            left_cst_translation = self.left.to_cst_translation(sql_access_method="")  #
+            if left_cst_translation.precode != []:
+                raise Exception("No support for prequeries in JoinNodes yet")  # TODO: Investigate this further...
```

Comment:
Yeah so this is where this whole dependency injection mess starts to suck.
For this we only need to check that there is no prequery (could also be supported in the future but probably out of scope...). This is why it does not matter, that the result is not a valid CSTStatement....

Same happens for the right join partner down below